cumulus/client/cirrus-executor/src/overseer.rs

```diff
@@ -365,187 +265,81 @@ pub async fn forward_events<PBlock, Client>(
-		active_leaves: HashMap<PBlock::Hash, NumberFor<PBlock>>,
-		overseer_config: CollationGenerationConfig<PBlock::Hash, NumberFor<PBlock>, Hash>,
-	) -> (Self, OverseerHandle<PBlock>) {
-		let (handle, events_rx) = mpsc::channel(SIGNAL_CHANNEL_CAPACITY);
-		let overseer = Overseer {
-			primary_chain_client,
-			overseer_config: Arc::new(overseer_config),
-			leaves,
-			active_leaves,
-			events_rx,
-		};
-		(overseer, OverseerHandle::new(handle))
-	}
+	let best_hash = primary_chain_client.info().best_hash;
 
-	/// Run the `Overseer`.
-	pub async fn run(mut self) -> Result<(), ApiError> {
-		// Notify about active leaves on startup before starting the loop
-		for (hash, number) in std::mem::take(&mut self.leaves) {
-			let _ = self.active_leaves.insert(hash, number);
-			let updated_leaf = ActivatedLeaf { hash, number };
-			if let Err(error) = self.on_activated_leaf(updated_leaf).await {
-				tracing::error!(
-					target: LOG_TARGET,
-					"Collation generation processing error: {error}"
-				);
-			}
-		}
-
-		while let Some(msg) = self.events_rx.next().await {
-			match msg {
-				Event::MsgToSubsystem(message) => {
-					if let Err(error) = self.handle_message(message).await {
-						tracing::error!(
-							target: LOG_TARGET,
-							"Collation generation processing error: {error}"
-						);
-					}
-				},
-				// TODO: we still need the context of block, e.g., executor gossips no message
-				// to the primary node during the major sync.
-				Event::BlockImported(block) => {
-					self.block_imported(block).await?;
-				},
-				Event::NewSlot(slot_info) =>
-					if let Err(error) = self.on_new_slot(slot_info).await {
-						tracing::error!(
-							target: LOG_TARGET,
-							"Collation generation processing error: {error}"
-						);
-					},
-			}
-		}
-
-		Ok(())
-	}
-
-	async fn on_activated_leaf(
-		&self,
-		activated_leaf: ActivatedLeaf<PBlock>,
-	) -> Result<(), ApiError> {
-		// follow the procedure from the guide
-		// TODO: invoke this on finalized block?
-		process_primary_block(
-			Arc::clone(&self.primary_chain_client),
-			&self.overseer_config,
-			(activated_leaf.hash, activated_leaf.number),
-		)
-		.await
-	}
+	let non_generic_best_hash =
+		PHash::decode(&mut best_hash.encode().as_slice()).expect("Hash type must be correct");
 
-	async fn on_new_slot(&self, slot_info: ExecutorSlotInfo) -> Result<(), ApiError> {
-		let client = &self.primary_chain_client;
-		let best_hash = client.info().best_hash;
+	let opaque_bundle = match bundler(non_generic_best_hash, executor_slot_info).await {
+		Some(opaque_bundle) => opaque_bundle,
+		None => {
+			tracing::debug!(target: LOG_TARGET, "executor returned no bundle on bundling",);
+			return Ok(())
+		},
+	};
 
-		let non_generic_best_hash =
-			PHash::decode(&mut best_hash.encode().as_slice()).expect("Hash type must be correct");
+	let () = primary_chain_client
+		.runtime_api()
+		.submit_transaction_bundle_unsigned(&BlockId::Hash(best_hash), opaque_bundle)?;
```

Comment:
`let () = `? 👀 

```suggestion
	primary_chain_client
		.runtime_api()
		.submit_transaction_bundle_unsigned(&BlockId::Hash(best_hash), opaque_bundle)?;
```

and a few other similar places.

Reply:
This is actually intentional. I know it looks odd, the idea was that in case those methods ever change signature, I want to make sure code stops compiling (it was unnecessarily returning `Option<()>` before).

I can remove it if you think it is not necessary.

Reply:
Nice to leave a comment somewhere as it's really confusing at the first glance and people might not able to get your intention on their own.

Personally, I don't think it's necessary as we usually won't touch it once it's merged. I don't remember why it's initially `Option<()>`, probably was copied without much thinking :D. I don't mind keeping it, just better avoid potential confusion.